### DISCUSSION TITLE
When Depends is not evaluated

### DISCUSSION THREAD

**Discussion by swsvc** [fastapi/fastapi#7337] [Questions] [answered]
Hi folks.

I think I'm missing something profound with Dependency injection concept in FastAPI.

Let's take this example: I have a function that is called somewhere inside the call chain of processing request and it has dependency. This is NOT the top-level api function which is usually marked by decorator like this `@router.get('/some_path')`

```
def some_callable_here():
     ...
     return some_value

async def do_something(some_var: str = Depends(some_callable_here)):
     # do something
```
My understanding of what happens here: if you don't pass value for `some_var` when calling `do_something`, this value gets injected using `Depends` by calling `some_callable_here`. 

My question: what do you absolutely need for this `Depends` to be evaluated? What I get in my is that `some_var` has value `Depends(...)` inside `do_something` which means that `some_callable_here` has never been evaluated. This question seems very abstract, but I'm lost and I don't know how to approach this problem. Maybe I can get to the answer myself if you point me to the part of code where `Depends` gets evaluated.

Thank you!

**Comment by swsvc** [comment, marked as answer]
Nevermind. In my case dependency injection won't work indeed. Hint for people who will struggle with the same problem: don't mix business logic with framework entities

**Comment by tiangolo** [comment]
Thanks for reporting back and closing the issue :+1:
